sdk/openai/azure-ai-openai-assistants/src/samples/java/com/azure/ai/openai/assistants/auth/ClientCreationWithNonAzureOpenAIKeyCredential.java

```diff
@@ -0,0 +1,100 @@
+
+        String deploymentOrModelId = "gpt-4-1106-preview";
+
+        AssistantsClient client = builder.buildAssistantsClient();
+//        AssistantCreationOptions assistantCreationOptions = new AssistantCreationOptions(deploymentOrModelId)
+//                .setFileIds(Arrays.asList("file-cQJ0ncMcsY7B8g6mio0RE102"))
+//                .setName("Shawn's Assistant")
+//                .setDescription("test-description")
+//                .setTools(Arrays.asList(new CodeInterpreterToolDefinition()));
+
+//        Assistant assistant = client.createAssistant(assistantCreationOptions);
+//        System.out.printf("Assistant ID=%s is created at %s.%n", assistant.getId(), assistant.getCreatedAt());
+
+        Assistant assistant = client.retrieveAssistant("asst_zS3eRcCrpOvbLltWEs6esGv1");
+        String name = assistant.getName();
+        System.out.println(name);
+
+        AssistantThreadsClient assistantThreadsClient = builder.buildAssistantThreadsClient();
+        AssistantThread assistantThread = assistantThreadsClient.retrieveThread("thread_Q9nzrJBH0lRs0gDqjV13npRJ");
+        System.out.println(assistantThread.getId());
+
+
+        ThreadMessagesClient threadMessagesClient = builder.buildThreadMessagesClient();
+        ThreadMessage message = threadMessagesClient.createMessage(assistantThread.getId(), MessageRole.USER, "Show me top 3 most profitable month.");
+        System.out.println(message.getContent().get(0));
+        ThreadMessage threadMessage = threadMessagesClient.retrieveMessage(assistantThread.getId(), message.getId());
+        System.out.println(threadMessage.getContent().get(0));
+
+
+        ThreadRunsClient threadRunsClient = builder.buildThreadRunsClient();
+        ThreadRun threadRun = threadRunsClient.createRun(assistantThread.getId(), "asst_zS3eRcCrpOvbLltWEs6esGv1");
+
+        String assistantId = threadRun.getAssistantId();
+        String threadId = threadRun.getThreadId();
+        String runId = threadRun.getId();
+        System.out.println("assistant id = " + assistantId + ", thread id = " + threadId + ", run id = " + runId);
+
+
+        ThreadRun threadRun1 = threadRunsClient.retrieveRun(threadId, runId);
+        System.out.println(threadRun1.getAssistantId());
+        while (threadRun1.getStatus() != RunStatus.COMPLETED) {
+            System.out.println(threadRun1.getStatus());
+            threadRun1 = threadRunsClient.retrieveRun(threadId, runId);
+        }
+
+
+//        RunStepsClient runStepsClient = builder.buildRunStepsClient();
+//        OpenAIPageableListOfRunStep openAIPageableListOfRunStep = runStepsClient.listRunSteps(threadId, runId);
+//        openAIPageableListOfRunStep.getFirstId();
+//        openAIPageableListOfRunStep.getLastId();
+
+
+//        OpenAIPageableListOfThreadMessage openAIPageableListOfThreadMessage = threadMessagesClient.listMessages(threadId);
+//
+//        List<ThreadMessage> data = openAIPageableListOfThreadMessage.getData();
```

Comment:
Currently, listing xxx APIs is wrong after using the new PR commit. They are non-public APIs but should be public APIs.

Reply:
Turn out it is intended in TypeSpec that marked as internal. 
https://github.com/Azure/azure-rest-api-specs/pull/27076/files#diff-ca640e72b094fdc36ef20a9975170a126ee737f82938914b6e02c2284a79e3cfR41-R43